integration-tests/tests/multichain/mod.rs

```diff
@@ -68,89 +48,11 @@ async fn test_triples_and_presignatures() -> anyhow::Result<()> {
-                .await?;
-            let tx_hash = ctx
-                .jsonrpc_client
-                .call(&RpcBroadcastTxAsyncRequest {
-                    signed_transaction: Transaction {
-                        nonce,
-                        block_hash,
-                        signer_id: signer.account_id().clone(),
-                        public_key: signer.public_key(),
-                        receiver_id: ctx.nodes.ctx().mpc_contract.id().clone(),
-                        actions: vec![Action::FunctionCall(FunctionCallAction {
-                            method_name: "sign".to_string(),
-                            args: serde_json::to_vec(&serde_json::json!({
-                                "payload": payload,
-                                "path": "test",
-                            }))?,
-                            gas: 300_000_000_000_000,
-                            deposit: 0,
-                        })],
-                    }
-                    .sign(&signer),
-                })
-                .await?;
-            tokio::time::sleep(Duration::from_secs(1)).await;
-            let is_tx_ready = || async {
-                let outcome_view = ctx
-                    .jsonrpc_client
-                    .call(RpcTransactionStatusRequest {
-                        transaction_info: TransactionInfo::TransactionId {
-                            hash: tx_hash,
-                            account_id: ctx.nodes.ctx().mpc_contract.id().clone(),
-                        },
-                    })
-                    .await?;
-                let FinalExecutionStatus::SuccessValue(payload) = outcome_view.status else {
-                    anyhow::bail!("tx finished unsuccessfully: {:?}", outcome_view.status);
-                };
-                let (big_r, s): (AffinePoint, Scalar) = serde_json::from_slice(&payload)?;
-                let signature = cait_sith::FullSignature::<Secp256k1> { big_r, s };
-                Ok(signature)
-            };
-            let signature = is_tx_ready
-                .retry(&ExponentialBuilder::default().with_max_times(6))
-                .await
-                .with_context(|| "failed to wait for signature response")?;
-            let mut bytes = vec![0x04];
-            bytes.extend_from_slice(&state_0.public_key.as_bytes()[1..]);
-            let point = EncodedPoint::from_bytes(bytes).unwrap();
-            let public_key = AffinePoint::from_encoded_point(&point).unwrap();
-            let epsilon = kdf::derive_epsilon(&account_id, "test");
-
-            assert!(signature.verify(
-                &kdf::derive_key(public_key, epsilon),
-                &Scalar::from_bytes(&payload),
-            ));
-
-            Ok(())
+            wait_for::has_at_least_triples(&ctx, 2).await?;
+            wait_for::has_at_least_presignatures(&ctx, 2).await?;
+            actions::single_signature_production(&ctx, &state_0).await
```

Comment:
It feels too much for me. This top-level approach may not give us enough flexibility. I would rather move the code from this particular function to the test itself. We may add such an abstraction if we have a few more tests.

Reply:
we can always degranularize the individual parts of the function. I already have a need for just running a single signature generation multiple times, so feels like this is a very common case when we want to test things on top of signature production